src/GoTrueClient.ts

```diff
@@ -634,20 +636,24 @@ export default class GoTrueClient {
    */
   private async _recoverAndRefresh() {
     try {
-      const json = isBrowser() && (await this.localStorage.getItem(STORAGE_KEY))
-      if (!json) {
-        return null
-      }
-
-      const data = JSON.parse(json)
+      const data = await getItemAsync(this.localStorage, STORAGE_KEY)
+      if (!data) return null
       const { currentSession, expiresAt } = data
       const timeNow = Math.round(Date.now() / 1000)
 
-      if (expiresAt < timeNow) {
+      if (expiresAt < timeNow + EXPIRY_MARGIN) {
```

Comment:
a counter and stop pinging gotrue or stop pinging after expiry time is reached cos gotrue is not going to refresh that token anyway

Reply:
Added exponential backoff, starting with 200ms and then exponentially backing off for 10 retries